src/main/java/org/opensearch/knn/index/util/AbstractKNNLibrary.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.util;
+
+import org.opensearch.common.ValidationException;
+import org.opensearch.knn.index.KNNMethod;
+import org.opensearch.knn.index.KNNMethodContext;
+
+import java.util.Map;
+
+/**
+ * AbstractKNNLibrary implements common functionality shared between libraries
+ */
+public abstract class AbstractKNNLibrary implements KNNLibrary {
+
+    protected final Map<String, KNNMethod> methods;
+    protected final String version;
+
+    /**
+     * Constructor
+     *
+     * @param methods Map of k-NN methods that the library supports
+     * @param version String representing version of library
+     */
+    AbstractKNNLibrary(Map<String, KNNMethod> methods, String version) {
+        this.methods = methods;
+        this.version = version;
+    }
+
+    @Override
+    public String getVersion() {
+        return this.version;
+    }
+
+    @Override
+    public KNNMethod getMethod(String methodName) {
+        return this.methods.get(methodName);
+    }
```

Comment:
please check if both getters can be done with Lombok @Getter at class level, not sure if that works with override and if we're not overriding those in sub-classes

Reply:
Not sure how to do getter for an item in a map, but can add for version.

Reply:
Oh, my bad, we do additional value by key lookup. Then single getter for "version" is the way to go